sip/lib/sip/outgoing_call.ex

```diff
@@ -0,0 +1,290 @@
+
+    message =
+      Sippet.Message.build_request(:invite, to_string(state.callee))
+      |> Map.put(:headers, headers)
+      |> Sippet.Message.put_header(:content_type, "application/sdp")
+      |> Map.replace(:body, proposal)
+
+    Call.make_request(message, state)
+  end
+
+  @impl Call
+  def handle_response(:invite, provisional, _response, state) when provisional in 100..199 do
+    case provisional do
+      100 -> notify_endpoint(state.endpoint, :trying)
+      180 -> notify_endpoint(state.endpoint, :ringing)
+      # Session Progress
+      183 -> nil
+      _xx -> Logger.warning("SIP Client: Unknown provisional response #{provisional}. Ignoring")
+    end
+
+    state
+  end
+
+  @impl Call
+  def handle_response(:invite, 200, response, state) do
+    state = send_ack(response, state)
+
+    case SDP.parse(response.body) do
+      {:ok, connection_info} ->
+        notify_endpoint(state.endpoint, {:call_ready, connection_info})
+        state
+
+      {:error, reason} ->
+        _state = build_and_send_request(:bye, state)
+
+        # Give Sippet time to send the request (this is async)
+        Process.sleep(20)
+
+        raise "SIP Client: Received SDP answer is not matching our requirements: #{inspect(reason)}"
+    end
+  end
+
+  @impl Call
+  def handle_response(:invite, transfer, response, state) when transfer in [300, 301, 302] do
+    case Sippet.Message.get_header(response, :contact, []) do
+      [{"Transfer", uri, _map} | _] ->
+        after_init(%{state | callee: uri})
+
+      _other ->
+        raise "SIP Client: Received #{transfer} response, but were unable to parse transfer URI"
+    end
+  end
+
+  @impl Call
+  def handle_response(:invite, declined, _response, state) when declined in [403, 603] do
+    notify_endpoint(state.endpoint, {:end, :declined})
+    state
+  end
+
+  @impl Call
+  def handle_response(:invite, busy, _response, state) when busy in [486, 600] do
+    notify_endpoint(state.endpoint, {:end, :busy})
+    state
+  end
+
+  @impl Call
+  def handle_response(:invite, 487, _response, state) do
+    # Request Terminated -- this is the case when the INVITE was cancelled
+    # by a separate CANCEL request before receiving a final response
+    state
+  end
+
+  @impl Call
+  def handle_response(:invite, 204, _response, _state) do
+    # 204 is a success response, but we need SDP from the response body to setup the session
+    # This shouldn't ever happen, but if we don't handle it here it would get silently ignored
+    raise "SIP Client: Received 204 No Content in response to INVITE"
+  end
+
+  @impl Call
+  def handle_response(method, 200, _response, state) when method in [:cancel, :bye] do
+    send(self(), :die)
+    state
+  end
+
+  @impl Call
+  def handle_response(_method, status_code, response, state) do
+    Call.handle_generic_response(status_code, response, state)
+  end
+
+  @impl Call
+  def handle_request(:notify, request, state) do
+    respond(request, 200)
```

Comment:
I'm not sure if I agree. I believe 200 is a well-known general success response -- plus, it would look weird if we just had this one status code as a module attribute, while other, more obscure SIP responses like 183 or 487, remained unannotated (you could argue that there are comments in code explaining these responses, but adding a module attribute for one response only seems like an inconsistency)

I think we should either:
* create module attributes for *every* response, e.g. 
```
@response_trying 100
@response_ringing 180
@response_session_progress 183
@response_ok 200
# etc
```
* leave these 200s as-is

Reply:
I think that creating module attributes for each response code sounds good because I am sure that I won't remember after 6 months that 180 is the status code for ringing.